### PR TITLE
feat: componetize event signing

### LINKED ISSUE #12174: Componentize event signing
Signing of a node's self events is currently handled by a method on `SwirldsPlatform.java`. We should turn self event signing into a formal platform component. This is not to improve performance. Rather, it will allow us to more cleanly mock self event signing during TURTLE style tests.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #12174

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java [hunk 1]
```diff
@@ -176,26 +175,17 @@ public BaseEventUnhashedData getUnhashedData() {
...
 
     /**
-     * Build the descriptor of this event. This cannot be done when the event is first instantiated, it needs to be
-     * hashed before the descriptor can be built.
+     * Ensure that the event descriptor has been built This cannot be done when the event is first instantiated, it
+     * needs to be hashed before the descriptor can be built.
      *
      * @throws IllegalStateException if the descriptor has already been built
      */
     public void buildDescriptor() {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java:187]: method is unused, and a duplicated of `getDescriptor`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/signing/DefaultSelfEventSigner.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public GossipEvent signEvent(@NonNull final BaseEventHashedData event) {
+        final Signature signature = new PlatformSigner(keysAndCerts).sign(event.getHash());
+        final BaseEventUnhashedData unhashedData = new BaseEventUnhashedData(null, signature.getSignatureBytes());
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/signing/DefaultSelfEventSigner.java:51]: Why not populate `otherId` with the correct value?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java [hunk 3]
```diff
@@ -106,6 +106,7 @@ public record PlatformSchedulersConfig(
         @ConfigProperty(defaultValue = "500") int consensusEngineUnhandledCapacity,
         @ConfigProperty(defaultValue = "SEQUENTIAL") TaskSchedulerType eventCreationManagerSchedulerType,
         @ConfigProperty(defaultValue = "500") int eventCreationManagerUnhandledCapacity,
+        @ConfigProperty(defaultValue = "DIRECT") TaskSchedulerConfiguration selfEventSigner,
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java:109]: missing doc for new param

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java [hunk 4]
```diff
@@ -610,6 +619,7 @@ public void wireExternalComponents(
      * @param platformPublisher         the platform publisher to bind
      */
     public void bind(
+            @NonNull final PlatformComponentBuilder builder,
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:622]: I'm assuming this PR represents the first step towards the pattern of passing *only* the `builder` into this method?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/tipset/TipsetUtils.java [hunk 5]
```diff
@@ -31,34 +30,15 @@ public final class TipsetUtils {
...
-            throw new IllegalStateException("event is not hashed");
-        }
-        return event.getHashedData().createEventDescriptor();
-    }
-
     /**
      * Get the descriptors of an event's parents.
      *
-     * @param event the event to get parent descriptors for
+     * @param event the event to the parent descriptors for
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/tipset/TipsetUtils.java:36]: grammared

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/tipset/TipsetUtils.java [hunk 6]
```diff
@@ -31,34 +30,15 @@ public final class TipsetUtils {
...
-    public static List<EventDescriptor> getParentDescriptors(@NonNull final EventImpl event) {
-        final List<EventDescriptor> parentDescriptors = new ArrayList<>(2);
-        if (event.getSelfParent() != null) {
-            parentDescriptors.add(buildDescriptor(event.getSelfParent()));
-        }
-        if (event.getOtherParent() != null) {
-            parentDescriptors.add(buildDescriptor(event.getOtherParent()));
-        }
-        return parentDescriptors;
+    public static List<EventDescriptor> getParentDescriptors(@NonNull final GossipEvent event) {
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/tipset/TipsetUtils.java:40]: Nope. Removed remaining uses and removed method.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java:187]: removed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/signing/DefaultSelfEventSigner.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/signing/DefaultSelfEventSigner.java:51]: unhashedData

Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/signing/DefaultSelfEventSigner.java:51]: We never should have written the other parent ID to this location in the first place. We can't remove the field in the short term due to the overhead of data migration, but that doesn't mean we need to write anything to it. Anything that still reads this field should be considered now and going forward.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java:109]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:622]: Now that I have merged develop, there are now 6 things we are binding via the builder.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/signing/DefaultSelfEventSigner.java (hunk 2 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/signing/DefaultSelfEventSigner.java:51]: Gotcha

**Review by alittley — APPROVED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by alittley — APPROVED** [review]

**Pull request merged**